script/grant_revoke_gem_authority.rb

```diff
@@ -1,166 +1,142 @@
-  KNOWN_MISMATCHED_GEM_NAMES.fetch(name, name)
-end
+@definitive_committers = File.read(
+  File.expand_path("../../data/definitive-rubygems-authors-list.txt", __FILE__)
+).split("\n").map { |line| line.sub(/ *#.*$/, "") }.select { |line| !line.empty? }
 
 ORGANIZATION_NAMES.each do |org_name|
   github.repos.list(org: org_name).each do |repo|
-    name = replace_known_mismatch(repo.name)
-    if exists?(name)
+    # Replace known mismatched names
+    name = KNOWN_MISMATCHED_GEM_NAMES.fetch(repo.name, repo.name)
+
+    # Let's move on to any gem that we know will be a false positive.
+    next if FALSE_POSITIVES.include?(name)
+
+    # TODO: We are twice calling `gem owner` for information. We could
+    # rework the logic to only call it once.
+    sleep(GEM_SLEEP_INTERVAL)
+    if system("gem owner #{name} > /dev/null 2>&1")
       @gem_names << name
     else
       @bogus_gem_names << repo.full_name
     end
   end
 end
 
-def gem_owner_with_error_check(gemname, params)
-  command = "gem owner #{gemname} #{params}"
-  puts "running: #{command}" if VERBOSE
+OWNERSHIP_ACTIONS = {
+  add: '-a',
+  remove: '-r'
+}
+
+# @param gemname [String]
+# @param profile [String] either an email associated with Rubygems or the
+#                         Rubygems.org profile name
+# @param action [Symbol] either :add or :remove (see OWNERSHIP_ACTIONS)
+def modify_gem_ownership(gemname:, profile:, action:)
+  switch = OWNERSHIP_ACTIONS.fetch(action)
+  sleep(GEM_SLEEP_INTERVAL)
+  command = "gem owner #{gemname} #{switch} #{profile}"
+  $stdout.puts "running: #{command}" if VERBOSE
   Open3.popen3(command) do |stdin, stdout, stderr, wait_thr|
-    @errors << "#{gemname} #{params}: #{stdout.read.chomp}" unless wait_thr.value.success?
+    @errors << "#{command}: #{stdout.read.chomp}" unless wait_thr.value.success?
   end
 end
 
-@gem_names.reject { |gemname| FALSE_POSITIVES.include?(gemname) }.sort.each do |gemname|
+
+@gem_names.sort.each do |gemname|
+  $stdout.puts "Updating #{gemname}..."
+
+  sleep(GEM_SLEEP_INTERVAL)
   current_committers = `gem owner #{gemname} | grep -e ^-`.split("\n")
   current_committers.collect! { |cc| cc.sub(/^.\s+/,'')}
 
+  committers_to_add = @definitive_committers - current_committers
+  committers_to_add.each do |profile|
+    modify_gem_ownership(gemname: gemname, profile: profile, action: :add)
+  end
+
   if ENV.fetch('WITH_REVOKE', false)
     committers_to_remove = current_committers - committer_emails
-    remove_params = committers_to_remove.map { |email| "-r #{email}" }.join(' ')
-    gem_owner_with_error_check(gemname, remove_params)
+    committers_to_remove.each do |profile|
+      modify_gem_ownership(gemname: gemname, profile: profile, action: :remove)
+    end
   end
-
-  committers_to_add = committer_emails - current_committers - SKIP_EMAILS
-  add_params = committers_to_add.map { |email| "-a #{email}" }.join(' ')
-  gem_owner_with_error_check(gemname, add_params)
 end
 
 if @bogus_gem_names.any?
```

Comment:
I wonder if it's worth spitting this out. If not, can also remove an extra `gem owner` call above, possibly? It seemed a good idea at some point, buuuuuut...